### PR TITLE
Added async validation to the templates to check the thesauri exists

### LINKED ISSUE #1003: Selecting a deleted thesauri on a template and saving it, should return a server error saying the thesauri no longer exists (5 pts)
- Right now it can happen that you select a thesauri that has been just deleted and the server saves it no problem, comming back to the template results in an error when trying to edit the property with the unexistent thesauri.
- Make sure the async validation happens on the "model" and not in the route.

### PR DESCRIPTION
**Pull request by konzz**
fixes #1003 

PR checklist:
- [ ] Update READ.me ?
- [ ] Update API documentation ?

QA checklist:
- [ ] Smoke test the functionality described in the issue
- [ ] Test for side effects
- [ ] UI responsiveness
- [ ] Cross browser testing
- [ ] Code review

### CODE REVIEW

File: app/api/templates/specs/fixtures.js [hunk 1]
```diff
@@ -14,6 +14,7 @@ const swapTemplate = db.id();
 const relatedTo = db.id();
 const templateToBeInherited = db.id();
 const propertyToBeInherited = db.id();
```
Comment (mfacar) [inline app/api/templates/specs/fixtures.js:16]: Please, can you remove the first six lines in this file given there aren't needed already?

File: app/shared/types/templateSchema.ts [hunk 2]
```diff
@@ -75,14 +76,20 @@ ajv.addKeyword('uniquePropertyFields', {
...
       return true;
     }
     if (['multiselect', 'select'].includes(data.type)) {
-      return !!(data.content && data.content.length);
+      if (!data.content || !data.content.length) {
+        return false;
+      }
+
+      const found = await thesauri.getById(data.content);
+      return !!found;
```
Comment (mfacar) [inline app/shared/types/templateSchema.ts:92]: Given the description, in the issue, this validation should throw a specific message "the thesaurus no longer exists", is it possible to change the definition of the keyword to return a ValidationError with a custom message?

**Review by mfacar — CHANGES_REQUESTED** [review]
LGTM!
I just have a doubt respect the message, please review the comments.

File: app/api/templates/specs/fixtures.js (hunk 1 above)
Comment (konzz) [reply app/api/templates/specs/fixtures.js:10]: done

File: app/shared/types/templateSchema.ts (hunk 2 above)
Comment (konzz) [reply app/shared/types/templateSchema.ts:92]: This error is quite rare, I think adding a different error message for this scenario only adds complexity and no real benefit, but thats just my opinion, so I will comment this today in the standup and see if we get an agreement

Comment (mfacar) [reply app/shared/types/templateSchema.ts:92]: For the record, we agreed to change the keyword to requireOrInvalidContentForSelectFields, having "validation failed: .properties[0] should pass "requireOrInvalidContentForSelectFields" keyword validation" as message

**Pull request merged**